app/views/unattended/provisioning_templates/finish/preseed_default_finish.erb

```diff
@@ -39,6 +39,11 @@ description: |
 # Install the root password
 echo 'root:<%= root_pass -%>' | /usr/sbin/chpasswd -e
 
+<%# Set hostname for Ubuntu Autoinstall: https://projects.theforeman.org/issues/34908 %>
+<% if @host.operatingsystem.name == "Ubuntu" && @host.operatingsystem.major.to_i > 20 || (@host.operatingsystem.major.to_i == 20 && @host.operatingsystem.minor.to_i >= 3) -%>
+hostnamectl set-hostname <%= @host.name %>
```

Comment:
Does the installer run systemd? In the past, it didn't, and you'd end up with errors like:

```console
# hostnamectl set-hostname lol
System has not been booted with systemd as init system (PID 1). Can't operate.
Failed to create bus connection: Host is down
```

Another question: in the `unless dhcp` section above, there is the following code:
```
/bin/echo <%= @host.shortname %> > /etc/hostname
/bin/hostname  <%= @host.shortname %>.<%= @host.domain %>
```

Which I would have expected is sufficient and would probably just need to be outside the `unless dhcp` block?

Reply:
Hey! Thanks for your response. 
I'd run the template on an Ubuntu 20.04.3 and Ubuntu 22.04 - both were able to set the hostname via `hostnameclt`, so I assume the installer runs systemd.

I'm gonna test the approach you are suggesting with moving the code block out of the `unless dhcp`.

Reply:
Tested the updated code & works fine.
Are you okay with the changes @evgeni ?